### ISSUE TITLE
Compiler error when using REWRITE

### ISSUE THREAD

**Issue by XYUnknown** [agda/agda#4452]
There is an error when I try to define `padCst`. The error message is:
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/TypeChecking/Rewriting.hs:378

The definition of `padCst` is:
```agda
  +zero : {m : ℕ} → m + zero ≡ m
  +zero {zero}  = refl
  +zero {suc m} = cong suc +zero

  +suc : {m n : ℕ} → m + (suc n) ≡ suc (m + n)
  +suc {m} {n} =
    begin
      m + (1 + n)
    ≡⟨ sym (+-assoc m 1 n) ⟩
      m + 1 + n
    ≡⟨ cong (_+ n) (+-comm m 1) ⟩
      refl

  {-# REWRITE +zero +suc #-}

  padCst : {n : ℕ} → (l r : ℕ) → {t : Set} → t → Vec t n → Vec t (l + n + r)
  padCst zero zero x xs = xs
  padCst zero (suc r) x xs = padCst zero r x (xs ++ [ x ])
  padCst (suc l) zero x xs = padCst l zero x ([ x ] ++ xs)
  {- the bug occurs here -}
  padCst (suc l) (suc r) x xs = padCst l r x ([ x ] ++ xs ++ [ x ])
```
The link of source code: [link](https://github.com/XYUnknown/individual-project/blob/master/src/example/PadCstIssue.agda)

**Comment by sattlerc** [comment]
Which version of Agda are you using? I don't get an internal error with the development version (master branch).

**Comment by XYUnknown** [comment]
I'm using Agda v2.6.0.1 with standard library agda-stdlib-1.2.

> Which version of Agda are you using? I don't get an internal error with the development version (master branch).

**Comment by andreasabel** [comment]
This is already fixed in the current development version (2.6.1).

Thanks for reporting!
